Update VariableTypeManual.cpp to not use catchAllKernel.

ailzhang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46354 Remove catchAllKernel_.
* #45939 Move catchAll to Math
* **#46353 Update VariableTypeManual.cpp to not use catchAllKernel.**

Differential Revision: [D24319416](https://our.internmc.facebook.com/intern/diff/D24319416)

Commit:
Update VariableTypeManual.cpp to not use catchAllKernel.

[ghstack-poisoned]

Commit:
Update on "Update VariableTypeManual.cpp to not use catchAllKernel."


Differential Revision: [D24319416](https://our.internmc.facebook.com/intern/diff/D24319416)

[ghstack-poisoned]

albanD commented on torch/csrc/autograd/VariableTypeManual.cpp:
```diff
@@ -334,27 +334,27 @@ static auto registry = torch::RegisterOperators()
     //      and remove the backend Autograd kernel here, only leaving the catch-all kernel.
     .kernel<decltype(VariableType::backward), &VariableType::backward>(DispatchKey::Autograd)
-    .catchAllKernel<decltype(VariableType::backward), &VariableType::backward>())
+    .kernel<decltype(VariableType::backward), &VariableType::backward>(DispatchKey::DefaultBackend))
```
Why is this one and requires_grad_ below on a different Key?

ailzhang replied:
Ah because they have Autograd kernels, registering to Math with Autograd will override Autograd, so we register it to DefaultBackend to use them only in inference.

albanD replied:
Thanks for the reference! Interesting.

ezyang reviewed (approved):
Can we port these to use the new registration API?

ailzhang commented:
Yea porting to new API should be easy, I'll update this PR!

Commit:
Update on "Update VariableTypeManual.cpp to not use catchAllKernel."


Differential Revision: [D24319416](https://our.internmc.facebook.com/intern/diff/D24319416)

[ghstack-poisoned]

bhosmer reviewed (commented):
oops just saw you landed this already - leaving these comments for whenever, no followup needed 😁
